src/biokbase/narrative/jobs/appmanager.py

```diff
@@ -466,41 +429,281 @@ def run_app(
             return new_job
 
     @_app_error_wrapper
-    def run_app_bulk(self, app_info, cell_id=None, run_id=None, dry_run=False):
+    def run_app_bulk(
+        self,
+        app_info: list,
```

Comment:
A TODO here might be to factor this into a list of AppInfo objects that validate themselves instead of bare dicts. That might make the front end slightly more complex, but not terribly so.

Then use the same idiom as an option in run_app? Maybe? Keeping things backward compatible is tricky.

Reply:
Note that this TODO should be completed before final release.